### ISSUE TITLE
ffmpeg_extract_subclip returns black frames

### ISSUE THREAD

**Issue by ksharsha** [Zulko/moviepy#508]
I am trying to extract particular sub clips from videos and I see that when the duration of the sub clip is shorter (around 2- 3 seconds), only black frames are seen in the subclip. I guess this might be because ffmpeg is unable to locate the key frames. How do I deal with this?

**Comment by Zulko** [comment]
Hey - That's not enough informations and I've never seen this before, can you provide some code to reproduce the issue ? (and possibly a videofile for which the issue exists too)

**Comment by ksharsha** [comment]
Hi, I recreated the issue using this gist https://gist.github.com/ksharsha/f28ca2f019070389c38741fb90a077d1 . I am attaching the video file as well. You can see that the audio is still preserved however, the video is completely blank. This is the video that I am trying to extract the subclip from (tstart = 12.334277199999999, tend = 15.97166675)

**Comment by earney** [comment]
It seems like this might help..
[https://trac.ffmpeg.org/wiki/Seeking](https://trac.ffmpeg.org/wiki/Seeking)

when I try what is recommended by this link, I do get video, but the video seems to be a bit messed up (like a bad key frame).  I'm not very good with ffmpeg, so maybe someone else sees a solution?

**Comment by ksharsha** [comment]
@earney true, the problem is with this cmd = [get_setting("FFMPEG_BINARY"),"-y",
      "-i", filename,
      "-ss", "%0.2f"%t1,
      "-t", "%0.2f"%(t2-t1),
      "-vcodec", "copy", "-acodec", "copy", targetname]
I am not good with ffmpeg either but looks like the above cmd has to be changed.

**Comment by D-Chief** [comment]
Is this issue resolved? Is there any other way instead of using ffmpeg_extract_subclip to clip videos?

**Comment by landogardner** [comment]
Looks like PR #579 could fix this.

**Comment by gpotter2** [comment]
Hi !

I made https://github.com/Zulko/moviepy/issues/882 which is quite clear and well-documented. It seems if fell unoticed, but it does fix this issue, in a better way than #579

Regards

**Comment by ebenz99** [comment]
Updating to say I believe this may still be a problem. Using ffmpeg_extract_subclip for 2-3 second clips and continue to run into this issue.

**Comment by keikoro** [comment]
@ebenz99 Should we reopen this issue?

**Comment by ebenz99** [comment]
Up to you. I ended up just using the workaround I mentioned (removing some options and taking a quality hit), but it might be nice to not have to do that.

**Comment by jeff-hykin** [comment]
This is an issue for me, I'm using moviepy to automate the refining of a machine learning dataset and it needs all the quality it can get.

Even with the quality hit, I'm not sure how to remove the copy/encode options, as I'm not passing any copy/encode options to `ffmpeg_extract_subclip()`

**Comment by persas** [comment]
Any update on this issue @jeff-hykin ?? I am in the very same position as you were back at the time that you commented on this

**Comment by persas** [comment]
Which options did you remove @ebenz99 ??

**Comment by jeff-hykin** [comment]
Because most of my time segments were tiny (less than 1 second), and I couldn't sacrifice on quality, I currently use opencv to load all of the frames, skip to the needed subsection, pull out all the frames, and then create a new video with those frames.

My older method was a direct call to ffmpeg
```python
def cut_video(source, start, end, output_path):
    """
    This takes a
    - filepath to a video
    - a start and end time in seconds (float or int)
    - an output_path
    It will create a new video clip based on the start and end times without modifying the original
    """
    duration = end - start
    if duration <= 0:
        raise Exception("duration screwed up for "+source)
    # touch the original video so that the file system knows it has been used recently
    Path(source).touch()
    # perform the video cut
    call(["ffmpeg", "-i", str(source), "-ss", str(start), "-t", str(duration), "-async", "1", str(output_path), "-hide_banner", "-loglevel", "panic"])
```

**Comment by keikoro** [comment]
Reopening because several people commented since close / problem persists.

**Comment by dennymarcels** [comment]
I have also encountered this issue. Curiously, it only happens with some of my videos, even if they were films made with the same camera. Also, I noticed that if I request the problematic clip to be a little longer, it eventually renders, but is very accelerated (5x or more, subjectively evaluated).

**Comment by OsaAjani** [comment]
Thank you for your contributions and for reporting issues in this repository. With the release of v2, which introduces significant changes to the codebase and API, we’ve reviewed the backlog of open PRs and issues. Due to the length of the backlog and the likelihood that many of these are either fixed or no longer applicable, we’ve made the decision to close all previous PRs and issues.

If you believe that any of these are still relevant to the current version or if you'd like to reopen a related discussion, please feel free to create a new issue or pull request, referencing the old one.

Thank you for your understanding and continued support!
